models/DataObject/ClassDefinition/Data/Multiselect.php

```diff
@@ -645,6 +645,35 @@ public function resolveBlockedVars(): array
         return $blockedVars;
     }
 
+    /**
+     * Perform a sanity check on option values, as a comma in option value leads to silent data corruption when persisted, see #5010.
+     *
+     * This method is called in DataObject\ClassDefinition::save()
+     *
+     * @param DataObject\ClassDefinition $class
+     * @param array $params
+     */
+    public function classSaved($class, $params = [])
+    {
+        $optionsProvider = DataObject\ClassDefinition\Helper\OptionsProviderResolver::resolveProvider(
+            $this->getOptionsProviderClass(),
+            DataObject\ClassDefinition\Helper\OptionsProviderResolver::MODE_MULTISELECT
+        );
+        if ($optionsProvider) {
+            $context = [];
+            $context['fieldname'] = $this->getName();
+
+            $options = $optionsProvider->{'getOptions'}($context, $this);
```

Comment:
```suggestion
            $options = $optionsProvider->getOptions($context, $this);
```

Reply:
Agree, I copied that part 1:1 some from some lines above. Not sure why it's there in the first place, as there aren't any special characters in getter.